src/Sparrow/Json/UnmanagedWriteBuffer.cs

```diff
@@ -332,24 +374,21 @@ public void Write<T>(in T value) where T : unmanaged
             }
             else
             {
-                WriteUnlikely(value);
+                WriteUnlikely(in value);
             }
         }
 
-        private void WriteUnlikely<T>(in T value) where T: unmanaged
+        private void WriteUnlikely<T>(in T data) where T : unmanaged
         {
-            int count = Unsafe.SizeOf<T>();
-            AllocateNextSegment(count, true);
-            
-            var head = _head;
+            AllocateNextSegment(sizeof(T), true);
 
             // Update Segment invariants
-            *(T*)(head.Address) = value;
-            head.AccumulatedSizeInBytes += count;
-            head.Used += count;
+            var head = _head;
+            *(T*)(head.Address + head.Used) = data;
+            head.AccumulatedSizeInBytes += sizeof(T);
+            head.Used += sizeof(T);
         }
```

Comment:
@ayende It seems like I have found the bug https://github.com/ravendb/ravendb/pull/20385/files#diff-c0c683607bad23e4d0eb79608e8c1bab99a8f6bca40da6f0a8b33df45914222eR347 already and fixed it. That's why it looked familiar to me.